zavod/zavod/context.py

```diff
@@ -190,26 +192,49 @@ def fetch_resource(
         )
 
     def fetch_response(
-        self, url: str, headers: _Headers = None, auth: _Auth = None
+        self,
+        url: str,
+        headers: _Headers = None,
+        auth: _Auth = None,
+        method: str = "GET",
+        data: _Body = None,
     ) -> Response:
-        """Execute an HTTP GET request using the contexts' session.
+        """Execute an HTTP request using the contexts' session.
 
         Args:
             url: The URL to be fetched.
             headers: HTTP request headers to be included.
             auth: HTTP basic authorization username and password to be included.
-
+            method: The HTTP method to use for the request.
+            data: The data to be sent in the request body.
         Returns:
             A response object.
         """
-        self.log.debug("HTTP GET", url=url)
+        self.log.debug(f"HTTP {method}", url=url)
         timeout = (settings.HTTP_TIMEOUT, settings.HTTP_TIMEOUT)
-        response = self.http.get(
+
+        kwargs: Dict[str, Any] = {
+            "headers": headers,
+            "auth": auth,
+            "timeout": timeout,
+            "data": data,
+        }
+
+        # This mimics the allow_redirects login found in requests.sessions
+        if method in ["GET", "OPTIONS"]:
+            kwargs["allow_redirects"] = True
+        elif method == "HEAD":
+            kwargs["allow_redirects"] = False
+        elif method in ["POST", "PUT", "PATCH", "DELETE"]:
+            # Deliberately noop for the sake of explicitness
+            pass
+        else:
+            raise ValueError(f"Unsupported HTTP method: {method}")
```

Comment:
is this needed? How did this come about? I ask because I think 307 and 308 response automatically reuse the same method, while 302 and 301 switch to GET? And now we're always providing a "allow_redirects" rather than only providing one when the crawler has specifically done so.

Reply:
As I mentioned in the comment, this is exact defaults taken from the requests lib.
i.e if you do requests.get with no allow_redirects set, it will set it to False, if you do requests.head, it'll set it to False.
Check this out:
https://github.com/psf/requests/blob/main/src/requests/sessions.py#L593-L624

Reply:
Oh cool. I see you're adding it because session.request always defaults to True while session.head defaults to False.